src/mqt/bench/bqskit_helper.py

```diff
@@ -0,0 +1,453 @@
+        if params is None:
+            params = []
+        self.check_parameters(params)
+
+        half_theta = float(params[0]) / 2
+        beta = float(params[1])
+        cos = math.cos(half_theta)
+        sin = math.sin(half_theta)
+        return UnitaryMatrix(
+            [
+                [1, 0, 0, 0],
+                [0, cos, -1j * sin * exp(-1j * beta), 0],
+                [0, -1j * sin * exp(1j * beta), cos, 0],
+                [0, 0, 0, 1],
+            ],
+        )
+
+    def get_grad(self, params: RealVector = None) -> npt.NDArray[np.complex128]:
+        """
+        Return the gradient for this gate.
+
+        See :class:`DifferentiableUnitary` for more info.
+        """
+        if params is None:
+            params = []
+        self.check_parameters(params)
+
+        half_theta = params[0] / 2
+        d11 = d22 = -1 / 2 * np.sin(half_theta)
+        d21 = -1 / 2 * 1j * exp(1j * params[1]) * np.cos(half_theta)
+        d12 = -1 / 2 * 1j * exp(-1j * params[1]) * np.cos(half_theta)
+
+        d_wrt_theta = np.array(
+            [
+                [0, 0, 0, 0],
+                [0, d11, d12, 0],
+                [0, d21, d22, 0],
+                [0, 0, 0, 0],
+            ],
+            dtype=np.complex128,
+        )
+
+        d12 = -exp(-1j * params[1]) * np.sin(half_theta)
+        d21 = exp(1j * params[1]) * np.sin(half_theta)
+
+        d_wrt_beta = np.array(
+            [
+                [0, 0, 0, 0],
+                [0, 0, d12, 0],
+                [0, d21, 0, 0],
+                [0, 0, 0, 0],
+            ],
+            dtype=np.complex128,
+        )
+
+        return np.array(
+            [d_wrt_theta, d_wrt_beta],
+            dtype=np.complex128,
+        )
+
+
+# mypy type checking is ignored for the ECRGate class because
+# ConstantGate and QubitGate has type Any
+class ECRGate(ConstantGate, QubitGate):  # type: ignore[misc]
+    """
+    The echoed cross-resonance gate.
+
+    The ECR gate is given by the following unitary:
+
+    .. math::
+
+        \\frac{1}{\sqrt{2}}\begin{pmatrix}
+        0 & 1 & 0 & i \\\\
+        1 & 0 & -i & 0 \\\\
+        0 & i & 0 & 1 \\\\
+        -i & 0 & 1 & 0
+        \\end{pmatrix}
+    """
+
+    _num_qudits = 2
+    _qasm_name = "opaque ecr"
+    _sqrt2 = math.sqrt(2)
+    _utry = UnitaryMatrix(
+        [
+            [0, 1 / _sqrt2, 0, 1j / _sqrt2],
+            [1 / _sqrt2, 0, -1j / _sqrt2, 0],
+            [0, 1j / _sqrt2, 0, 1 / _sqrt2],
+            [-1j / _sqrt2, 0, 1 / _sqrt2, 0],
+        ]
+    )
+
+
+def get_rebase(gate_set: list[str]) -> list[Gate]:
+    op_dict = {
+        "rx": RXGate(),
+        "ry": RYGate(),
+        "rz": RZGate(),
+        "rxx": RXXGate(),
+        "rzz": RZZGate(),
+        "sx": SXGate(),
+        "x": XGate(),
+        "cp": CPGate(),
+        "cx": CNOTGate(),
+        "cz": CZGate(),
+        "ecr": ECRGate(),
+        "xx_plus_yy": XXPlusYYGate(),
```

Comment:
Could you also make sure that this gate is implemented for TKET as well in the `tket_helper.py` file? Same for the `cp` gate.